src/desktop/dialogs/settingsdialog/tools.cpp

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "desktop/dialogs/settingsdialog/tools.h"
+#include "desktop/scene/canvasview.h"
+#include "desktop/settings.h"
+#include "desktop/utils/sanerformlayout.h"
+
+#include <QEvent>
+#include <QPainter>
+#include <QPen>
+#include <QCheckBox>
+#include <QComboBox>
+#include <QDoubleSpinBox>
+#include <QVariant>
+#include <QWidget>
+#include <QtColorWidgets/ColorWheel>
+#include <limits>
+#include <utility>
+
+namespace dialogs {
+namespace settingsdialog {
+
+enum Driver {
+	WindowsInk,
+	WinTab,
+	WinTabRelativePenMode,
+};
+
+Tools::Tools(desktop::settings::Settings &settings, QWidget *parent)
+	: QWidget(parent)
+{
+	auto *form = new utils::SanerFormLayout(this);
+
+	initGeneralTools(settings, form);
+	form->addSeparator();
+	initColorWheel(settings, form);
+}
+
+void Tools::initGeneralTools(desktop::settings::Settings &settings, utils::SanerFormLayout *form)
+{
+	auto *toggleKeys = new QCheckBox(tr("Toggle between previous and current tool"));
+	settings.bindToolToggle(toggleKeys);
+	form->addRow(tr("Keyboard shortcuts:"), toggleKeys, 1, 2);
+
+	form->addSeparator();
+
+	auto *shareColor = new QCheckBox(tr("Share one color across all brush slots"));
+	settings.bindShareBrushSlotColor(shareColor);
+	form->addRow(tr("Brushes:"), shareColor, 1, 2);
+
+	auto *outlineSize = new QDoubleSpinBox;
+	settings.bindBrushOutlineWidth(outlineSize);
+	outlineSize->setDecimals(1);
+	outlineSize->setMaximum(25.0);
+	outlineSize->setSingleStep(.5);
+	outlineSize->setSuffix(tr("px"));
+	auto *outlineSizeLayout = utils::encapsulate(tr("Show a %1 outline around the brush"), outlineSize);
```

Comment:
I'm told that this encapsulated style of setting is confusing and difficult to translate to Chinese, and thinking about it, that's the case for German as well. You end up with really long and complex sentences where before a single compound noun suffices.